.gitattributes

```diff
@@ -0,0 +1 @@
+Makefile linguist-generated=true
```

Comment:
Not sure this can be applied to a whole repository - how do you know that there aren't sub-directories with non-generated Makefiles?

Reply:
That is a good point, didn't think of that scenario 😅 

I see two alternatives:
1. We keep this change, assume most users only have generated Makefiles, and make it more clear in documentation and `sage init` output what this implies. However that information might be forgotten when later introducing manually written Makefiles.
2. Only document suggestion to add this to your repo as a convenience.

I would prefer 1 to optimize for most users (assuming that is true), but understand if that is perhaps too bold and we should be more safe instead. WDYT?

Reply:
What if we wrote absolute paths to all generated Makefiles? Feels like the only waterproof solution

Reply:
Hmm, that would also make this propagate to existing repositories nicely! I'll see what I can do!

Reply:
Cool! Thinking out loud here but what if we added our own attribute aswell `sage-generated=true` to be able to "clean" out attributes for removed Makefiles?

Reply:
That is a good point. I don't see that we currently keep track of removed generated Makefiles anywhere, so I guess we'd have to compare the list of files-with-`sage-generated=true`-attributes with the list of generated makefiles and clean out the removed ones. 🤔 

I'll see if this is something that could easily be incorporated when I take a look at this PR again 🙏